### PR TITLE
Properly create secondary endpoint Uri for Azurite endpoints

### LINKED ISSUE #17215: The way of storage table second endpoint constructed in Azure SDK for net doesn't support use in Azurite
**Error Description:**
[Azurite](https://github.com/Azure/Azurite) is an Azure Storage emulator. 
Current storage table second endpoint which is constructed in Azure SDK for net can not access Azurite. The reason is the storage table second endpoint for Azurite is different from the one of an Azure storage account.
In Azurite, storage table second endpoint constructed is shown as following:
```
https://127.0.0.1:10002/<account-name>-secondary/<resource-path>
```
In Azure SDK for net, storage table second endpoint constructed is shown as following:
```
https://<account-name>-secondary.table.core.windows.net/<resource-path>
```

**Error Track:**
The code for storage table second endpoint constructed in Azure SDK for net is [here](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/tables/Azure.Data.Tables/src/TableServiceClient.cs#L151)
Screenshot 2020-11-26 163623

**Expected Behavior**
Storage table second endpoint which is constructed in azure sdk for net can access azurite.

@jongio for notification.

### PR DESCRIPTION
**Pull request by christothes**
fixes #17215

### CODE REVIEW

File: sdk/tables/Azure.Data.Tables/tests/TableConnectionStringTests.cs [hunk 1]
```diff
@@ -101,6 +106,23 @@ public void ParseFailsWithInvalidConnString(string connString)
...
+        }
+
+        [Test]
+        [TestCaseSource(nameof(ValidPrimaryEndpointUris))]
+        public void GetSecondaryUriFromPrimary(Uri endpoint)
+        {
+            Uri secondaryEndpoint = TableConnectionString.GetSecondaryUriFromPrimary(endpoint);
+
+            Assert.That(secondaryEndpoint, Is.Not.Null.Or.Empty, "Secondary endpoint should not be null or empty");
+            Assert.That(secondaryEndpoint.AbsoluteUri.IndexOf($"{AccountName}{TableConstants.ConnectionStrings.SecondaryLocationAccountSuffix}", StringComparison.OrdinalIgnoreCase), Is.GreaterThanOrEqualTo(0), $"Secondary endpoint should contain '{AccountName}-secondary', but was {secondaryEndpoint.AbsoluteUri}");
```
Comment (pakrym) [inline sdk/tables/Azure.Data.Tables/tests/TableConnectionStringTests.cs:123]: Let's have a stricter assert against a known expected value.

File: sdk/tables/Azure.Data.Tables/src/TableConnectionString.cs [hunk 2]
```diff
@@ -266,6 +269,44 @@ static Uri CreateUri(string endpoint, string sasToken)
...
+            }
+
+            var indexOfDot = secondaryUriBuilder.Host.IndexOf('.');
+            if (indexOfDot >= 0 && Uri.CheckHostName(primaryUri.Host) == UriHostNameType.Dns)
+            {
+                // This is a dns name such as contoso.core.windows.net
+                // Insert the '-secondary' suffix after the first part of the host name
+                secondaryUriBuilder.Host = secondaryUriBuilder.Host.Insert(indexOfDot, TableConstants.ConnectionStrings.SecondaryLocationAccountSuffix);
+            }
+            else if (primaryUri.IsLoopback)
```
Comment (pakrym) [inline sdk/tables/Azure.Data.Tables/src/TableConnectionString.cs:291]: Is it useful to have a secondary uri for the loopback?

**Review by pakrym — APPROVED** [review]

File: sdk/tables/Azure.Data.Tables/src/TableConnectionString.cs (hunk 2 above)
Comment (christothes) [reply sdk/tables/Azure.Data.Tables/src/TableConnectionString.cs:291]: I assume it matters to Azureite - see https://github.com/Azure/azure-sdk-for-net/issues/17215

File: sdk/tables/Azure.Data.Tables/tests/TableConnectionStringTests.cs (hunk 1 above)
Comment (christothes) [reply sdk/tables/Azure.Data.Tables/tests/TableConnectionStringTests.cs:123]: Probably not a bad idea.

**Pull request merged**